clash-lib/src/Clash/Core/Subst.hs

```diff
@@ -891,7 +902,10 @@ acmpTerm' inScope = go (mkRnEnv inScope)
     go env l1 l2 `thenCompare` go env r1 r2
   go env (TyApp l1 r1) (TyApp l2 r2) =
     go env l1 l2 `thenCompare` acmpType' env r1 r2
-  go env (Letrec bs1 e1) (Letrec bs2 e2) =
+  go env (Let (NonRec i1 x1) e1) (Let (NonRec i2 x2) e2) =
+    let env' = rnTmBndr env i1 i2
+     in go env' x1 x2 `thenCompare` go env' e1 e2
```

Comment:
Shouldn't this be:
```suggestion
    go env x1 x2 `thenCompare` go env' e1 e2
```
Since `i1` doesn't scope over `x1` and `i2` doesn't scope over `x2`.